engine/src/org/pentaho/di/trans/steps/monetdbbulkloader/MonetDbVersion.java

```diff
@@ -0,0 +1,140 @@
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+package org.pentaho.di.trans.steps.monetdbbulkloader;
+
+import static java.util.regex.Pattern.compile;
+
+import java.util.regex.Pattern;
+
+import org.pentaho.di.i18n.BaseMessages;
+
+/**
+ * @author Tatsiana_Kasiankova
+ * 
+ */
+public class MonetDbVersion implements Comparable<MonetDbVersion> {
+  private static Class<?> PKG = MonetDbVersion.class; // for i18n purposes, needed by Translator2!!
+
+  private Integer minorVersion;
+
+  private Integer majorVersion;
+
+  private Integer patchVersion;
+
+  private static final String SEPARATOR = "\\.";
+
+  private static final Pattern VERSION_PATTERN = compile( "^[0-9]+(\\.[0-9]+)*$" );
+
+  /**
+   * @param minorVersion
+   * @param majorVersion
+   * @param patchVersion
+   */
+  public MonetDbVersion( int minorVersion, int majorVersion, int patchVersion ) {
+    super();
+    this.minorVersion = minorVersion;
+    this.majorVersion = majorVersion;
+    this.patchVersion = patchVersion;
+  }
+
+  /**
+   * @param productVersion
+   */
+  public MonetDbVersion( String productVersion ) {
+    super();
+    parseVersion( productVersion );
+  }
+
+  /**
+   * @return the minorVersion
+   */
+  public Integer getMinorVersion() {
+    return minorVersion;
+  }
+
+  /**
+   * @return the majorVersion
+   */
+  public Integer getMajorVersion() {
+    return majorVersion;
+  }
+
+  /**
+   * @return the patchVersion
+   */
+  public Integer getPatchVersion() {
+    return patchVersion;
+  }
+
+  @Override
+  public int compareTo( MonetDbVersion mDbVersion ) {
+    int result = majorVersion.compareTo( mDbVersion.majorVersion );
+    if ( result != 0 ) {
+      return result;
+    }
+
+    result = minorVersion.compareTo( mDbVersion.minorVersion );
+    if ( result != 0 ) {
+      return result;
+    }
+
+    result = patchVersion.compareTo( mDbVersion.patchVersion );
+    if ( result != 0 ) {
+      return result;
+    }
+    return result;
+  }
+
+  private void parseVersion( String productVersion ) {
+    if ( productVersion == null ) {
+      throw new IllegalArgumentException( BaseMessages.getString( PKG, "MonetDBVersion.Exception.VersionIsNull" ) );
+    }
+
+    if ( !VERSION_PATTERN.matcher( productVersion ).matches() ) {
```

Comment:
The reason is we don't know exactly how many digits segments could be present in valid MonetDB version.
So we do an assumption that any sequence of digits segments separated with dots is valid MonetDB version.
In this case we can always compare current MonetDB version with boundary Jan2014-SP2=11.17.17
This allows to be more stable at checking Monet DB Version.

For example if we get version=11.17.17.6.25, we shouldn't stop checking and we can compare such a version with boundary Jan2014-SP2=11.17.17.